src/libs/Revision.cpp

```diff
@@ -0,0 +1,62 @@
+#include "checksumm.h"
+#include "Config.h"
+#include "ConfigValue.h"
+#include "libs/Pin.h"
+#include "libs/Kernel.h"
+#include "libs/utils.h"
+#include "Revision.h"
+
+
+#define pcb_revision_bit_0           CHECKSUM("pcb_revision_bit_0")
+#define pcb_revision_bit_1           CHECKSUM("pcb_revision_bit_1")
+
+static const char* REV12_NAME = "12";
+static const char* REVA_NAME = "A";
+static const char* REVB_NAME = "B";
+static const char* REVC_NAME = "C";
+static const char* REVINVALID_NAME = "INVALID";
+
+const char *string_from_rev(const Rev& r) {
+    switch (r) {
+        case REV_12:
+            return REV12_NAME;
+        case REV_A:
+            return REVA_NAME;
+        case REV_B:
+            return REVB_NAME;
+        case REV_C:
+            return REVC_NAME;
+        case REV_INVALID:
```

Comment:
Is it valid to have a case without a return statement

Reply:
yup, that's called "fallthrough". if you have a case without a `break` or `return`, execution moves to the next case. so what this is doing is treating `REV_INVALID` and `default` the same.